restli-server/src/main/java/com/linkedin/restli/internal/server/util/ArgumentUtils.java

```diff
@@ -530,6 +591,17 @@ public static Object convertSimpleValue(final String value,
       underlyingValue = ValueConverter.coerceString(value, DataSchemaUtil.dataSchemaTypeToPrimitiveDataSchemaClass(dereferencedType));
     }
 
+    if(validateKey)
+    {
+      DataSchemaAnnotationValidator validator = new DataSchemaAnnotationValidator(schema);
+      ValidationResult result = ValidateDataAgainstSchema.validate(underlyingValue, schema,
+          new ValidationOptions(RequiredMode.CAN_BE_ABSENT_IF_HAS_DEFAULT, CoercionMode.OFF), validator);
+      if(!result.isValid())
+      {
+        throw new RoutingException(String.format("Key with value '%s' is invalid, reason: %s",
```

Comment:
I am curious to know why we throw RoutingException here? Shall we keep the experience consistent with previous one. Throw IllegalArgumentException here, then the exception will be caught in the caller. Then the caller throws RoutingException?

Reply:
In ArgumentBuilder IllegalArgumentException is handled but this util method is used in RestliRouter which doesn't and handles RoutingException only.